render.c

```diff
@@ -274,6 +280,18 @@ void render_frame(struct swaylock_surface *surface) {
 			cairo_set_source_u32(cairo, state->args.colors.layout_text);
 			cairo_show_text(cairo, layout_text);
 			cairo_new_sub_path(cairo);
+
+			new_height += fe.height + 2 * box_padding;
```

Comment:
Hmm, is this correct? The buffer will grow vertically each time we render.

(IMHO we should probably set `new_width = new_height = 0` at the beginning of the function, always computing the new size without taking into account the old one)

Reply:
Yes this is correct. The compute of the new size doesn't take into account the old. Actually, its starting value is `buffer_diameter`. If the keymap indicator is to be rendered too, we need to take it into account and add it to the height.

Reply:
Oh, indeed, I got confused with `buffer_width/height` vs. `buffer_diameter`